inc/abstractrightsdropdown.class.php

```diff
@@ -0,0 +1,252 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+abstract class AbstractRightsDropdown
+{
+   /**
+    * Max limit per itemtype
+    */
+   const LIMIT = 50;
+
+   /**
+    * To be redefined by subclasses, specify enabled types
+    *
+    * @return array
+    */
+   protected static function getTypes(): array {
+      return [];
+   }
+
+   /**
+    * Check if a given type is enabled
+    *
+    * @param string $type Class to check
+    *
+    * @return bool
+    */
+   protected static function isTypeEnabled(string $type): bool {
+      $types = array_flip(static::getTypes());
+      return isset($types[$type]);
+   }
+
+   /**
+    * Get possible data for profiles
+    *
+    * @param string $name  Field name
+    * @param array $values Selected values
+    *
+    * @return array
+    */
+   public static function show(string $name, array $values): string {
+      global $CFG_GLPI;
+
+      // Build DOM id
+      $field_id = $name . "_" . mt_rand();
+
+      // Build url
+      $url = $CFG_GLPI['root_doc'] . "/ajax/getShareDashboardDropdownValue.php";
```

Comment:
You're in an abstract class, then the URL to the ajax resource should not mention dropdown

Maybe 
```suggestion
      $url = $CFG_GLPI['root_doc'] . "/ajax/getRightDropdownValue.php";
```

Reply:
Good catch, forgot about that.
Done 307ad74a2f.